### ISSUE TITLE
TAP output should be spec-compliant

### ISSUE THREAD

**Issue by elmarco** [avocado-framework/avocado#3317]
meson TAP parser is quite strict and fails with avocado output on error with failfast:

```
/home/elmarco/src/qq/tests/avocado-test.sh tests/venv x86_64 /home/elmarco/src/qq/tests/acceptance /home/elmarco/src/qq/build/tests/results
--- stdout ---
1..17
ok 1 /home/elmarco/src/qq/tests/acceptance/boot_linux_console.py:BootLinuxConsole.test_x86_64_pc
ok 2 /home/elmarco/src/qq/tests/acceptance/cpu_queries.py:QueryCPUModelExpansion.test
ok 3 /home/elmarco/src/qq/tests/acceptance/empty_cpu_model.py:EmptyCPUModel.test
ok 4 /home/elmarco/src/qq/tests/acceptance/linux_initrd.py:LinuxInitrd.test_with_2gib_file_should_exit_error_msg_with_linux_v3_6
not ok 5 /home/elmarco/src/qq/tests/acceptance/linux_initrd.py:LinuxInitrd.test_with_2gib_file_should_work_with_linux_v4_16
Interrupting job (failfast).
--- stderr ---

TAP parsing error: unexpected input at line 7
TAP parsing error: Too few tests run (expected 17, got 5)
(test program exited with status code 9)
```

'Interrupting job (failfast).' shouldn't be in the output, or with # comment.

**Comment by clebergnu** [comment]
@elmarco good point, this looks very much a like a bug.  I'll be working on a fix.  Thanks for reporting it.

**Comment by clebergnu** [comment]
@elmarco an attempt to fix this is on #3318.  Let me know if you happen to have the means to test it, or, share the meson script you're using.

**Comment by clebergnu** [comment]
@elmarco closing this assuming the fix was valid.  Please reopen if you find any shortcomings.  Thanks again!
